Wikipedia/Code/WMFThemeableNavigationController.m

```diff
@@ -34,30 +34,19 @@ - (UIStatusBarStyle)preferredStatusBarStyle {
 
 - (void)applyTheme:(WMFTheme *)theme {
     self.theme = theme;
-    self.navigationBar.barTintColor = theme.colors.chromeBackground;
     self.navigationBar.translucent = NO;
     self.navigationBar.tintColor = theme.colors.chromeText;
-    UIImage *backgroundImage = nil;
-    switch (self.style) {
-        case WMFThemeableNavigationControllerStyleEditor:
-            backgroundImage = theme.editorNavigationBarBackgroundImage;
-            break;
-        case WMFThemeableNavigationControllerStyleSheet:
-            backgroundImage = theme.sheetNavigationBarBackgroundImage;
-            break;
-        case WMFThemeableNavigationControllerStyleGallery:
-            self.modalTransitionStyle = UIModalTransitionStyleCrossDissolve;
-            [self setNavigationBarHidden:YES animated:NO];
-            break;
-        default:
-            backgroundImage = theme.navigationBarBackgroundImage;
-            break;
+    UINavigationBarAppearance *appearance = [UINavigationBarAppearance appearanceForTheme:theme style:self.style];
+    self.navigationBar.standardAppearance = appearance;
+    self.navigationBar.scrollEdgeAppearance = appearance;
+    self.navigationBar.compactAppearance = appearance;
+
+    if (@available(iOS 15.0, *)) {
+        //do nothing
+    } else {
```

Comment:
```suggestion
    if (@available(iOS 15.0, *) == false) {
```

Reply:
I see the same warning message here - `@available does not guard availability here; use if (@available) instead`.

Reply:
Thank you, good catch! (And sorry for not fully examining my suggestion.) We'll keep it as is. Great work on this.